packages/lesswrong/components/common/withDialog.tsx

```diff
@@ -23,26 +23,25 @@ export const OpenDialogContext = React.createContext<OpenDialogContextType|null>
 export const DialogManager = ({children}: {
   children: React.ReactNode,
 }) => {
-  const [componentName,setComponentName] = useState<CloseableComponents|null>(null);
-  const [componentProps,setComponentProps] = useState<any>(null);
+  const [componentNameWithProps, setComponentNameWithProps] = useState<{componentName: CloseableComponents, componentProps: any} | null>(null);
```

Comment:
This fixes a bug where when you load a new dialog, it would first load the dialog with the props of the previously loaded dialog component (or empty props if there are none). This is of course sad and bad, and this changes the coupling of the state so that that is no longer possible.